packages/synapse-sdk/src/telemetry/service.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * TelemetryService - Main telemetry service for Synapse SDK
+ *
+ * You should use `synapse.telemetry.sentry` directly to capture events.
+ *
+ * ## Debug Dumps
+ *
+ * Get recent events for support tickets:
+ *
+ * ```typescript
+ * const dump = synapse.telemetry.debugDump()
+ * console.log(JSON.stringify(dump, null, 2))
+ * ```
+ */
+
+import { resolveTelemetryConfig } from './config.ts'
+import { getSentry, type Sentry as SentryType } from './get-sentry.ts'
+
+export interface TelemetryConfig {
+  /**
+   * Whether to enable telemetry.
+   *
+   * You can also control this via the environment variable `SYNAPSE_TELEMETRY_DISABLED` or the global variable `SYNAPSE_TELEMETRY_DISABLED`.
+   *
+   * This value will also be false if `NODE_ENV === 'test'`.
+   *
+   * @default true
+   */
+  enabled?: boolean
+  /**
+   * The name of the application using the SDK.
+   * This is used to identify the application in the telemetry data.
+   * This is optional and can be set by the user via the synapse.telemetry.sentry.setContext() method.
+   * If not set, the SDK will use 'synapse-sdk' as the default app name.
+   */
+  appName?: string
+  tags?: Record<string, string> // optional: custom tags
+}
+
+/**
+ * Configuration for runtime detection and context
+ */
+interface RuntimeContext {
+  sdkVersion: string
+  runtime: 'browser' | 'node'
+  network: 'mainnet' | 'calibration'
+  userAgent?: string
+  appName?: string
+}
+
+export interface DebugDump {
+  lastEventId: string | undefined
+  events: any[]
+}
+
+const { Sentry, integrations } = await getSentry()
+
+/**
+ * Main telemetry service that manages the adapter and provides high-level APIs
+ */
+export class TelemetryService {
+  // private adapter: TelemetryAdapter
+  private enabled: boolean
+  private context: RuntimeContext
+  private eventBuffer: any[] = []
+  private readonly maxBufferSize = 50
+
+  readonly sentry: SentryType
+
+  constructor(config: TelemetryConfig, context: RuntimeContext) {
+    const resolvedConfig = resolveTelemetryConfig(config)
+    this.enabled = resolvedConfig.enabled
+    this.context = context
+
+    this.sentry = Sentry
+    if (this.enabled) {
+      this.initSentry()
+    }
+  }
+  private initSentry(): void {
+    this.sentry.init({
+      dsn: 'https://3ed2ca5ff7067e58362dca65bcabd69c@o4510235322023936.ingest.us.sentry.io/4510235328184320',
+      // Setting this option to false will prevent the SDK from sending default PII data to Sentry.
+      // For example, automatic IP address collection on events
+      sendDefaultPii: false,
+      release: `@filoz/synapse-sdk@v${this.context.sdkVersion}`,
+      beforeSend: this.sanitizeError.bind(this),
```

Comment:
```suggestion
      beforeSend: this.addToBuffer.bind(this),
```

Doesn't seem to be any point adding extra indirection now

Reply:
see [`07b6cbc` (#349)](https://github.com/FilOzone/synapse-sdk/pull/349/commits/07b6cbc711fa9a0e30a03f4b0b2727da7bcc9838)